src/ccpp_types.F90

```diff
@@ -19,6 +19,10 @@
 !
 module ccpp_types
 
+#ifdef MPI
+    use mpi_f08, only: MPI_Comm
+#endif
```

Comment:
Do we still need this `#ifdef`?
(question applies to all use of `#ifdef MPI`)

Reply:
The idea was that we want to be able to build without MPI altogether. In this case, the changes in this PR define a fake MPI_Comm. The interfaces to the physics don't have to change this way as long as the physics use `mpi_comm` as defined (no mpi) or re-exported (mpi_f08) by the framework.

Reply:
What happens in the physics if it executes MPI commands (e.g., `MPI_allgather` at timestep init time) but MPI is faked?

Reply:
All these MPI calls need to be in ifdefs unfortunately - unless someone volunteers to write an entire stub MPI implementation. If we want to go down that route. If we say however that we make MPI a strict requirement for CCPP framework and physics, then we don't need to worry. For most of our models, MPI is needed anyways, but for example the Single Column Model doesn't use it - it's only one column.

Reply:
You mean that all CCPP physics routines that use MPI also need these `ifdefs`?
How can they function correctly? You can't just `ifdef` out a call to `MPI_allgather` and expect correct behavior. I feel like I'm missing something here.

For a "stub" (really serial) version of MPI, have a look at https://github.com/ESMCI/mpi-serial
Is that what you are talking about?

Reply:
So this is what we need to decide - do we want to be able to exercise CCPP without MPI or make it a hard requirement. I am fine either way. If it's a hard requirement, we have to make this clear in the documentation. Users like the SCM will then have to use MPI with only one mpi task. Since SCM uses spack-stack, MPI is available and loaded anyway, because hdf5/netcdf are built with parallel I/O enabled.

Reply:
and yes, correct, for some of the mpi calls (that are not bcast for example) you need to put alternative code or use the ESMCI mpi-serial or something similar. Rather messy and maybe not worth the effort?

Reply:
I feel like we should make some policy statement about this. I am not sure if it should be informational or an outright standard for the CCPP.
This statement should clarify whether MPI is assumed to be available for CCPP physics schemes.
@peverwhee, any objections to the `#ifdef MPI` approach? CESM is always built with MPI but this would make schemes (currently only QNEG I think) 'portable'.

Reply:
We can certainly implement the `#ifdef` if the wider community prefers that, or if it turns out to be difficult to implement MPI in models like SCM, but I think from the SIMA viewpoint we might lean more towards just making MPI a host-model requirement.  

This is partially because as @gold2718 pointed out we are always building with MPI already (even when in single-column mode), partially because `#ifdef` can make the code harder to read and reduce runtime configurability, and partially because I am a little wary about being able to fully replicate the behavior of the MPI calls with our own personally-developed serial call, which means the schemes could have different behavior even if they are "portable".

Also I should note that most of our core, likely more scientifically-interesting, schemes don't directly use MPI at all, and so they should remain portable regardless.

Of course this is just the view from our side of things, and so we will be happy to implement the `ifdef` method if it makes it easier for NOAA, NRL, or other host model groups.